docker/boost-client/Dockerfile.source

```diff
@@ -0,0 +1,73 @@
+#########################################################################################
+FROM golang:1.20-bullseye as builder
+
+RUN apt update && apt install -y \
+      build-essential \
+      bzr pkg-config \
+      clang \
+      curl \
+      gcc git \
+      hwloc \
+      jq \
+      libhwloc-dev wget \
+      mesa-opencl-icd \
+      ocl-icd-opencl-dev
+
+WORKDIR /go/src/
+
+COPY Makefile /go/src/
+### make configurable filecoin-ffi build
+ARG FFI_BUILD_FROM_SOURCE=0
+ENV FFI_BUILD_FROM_SOURCE=${FFI_BUILD_FROM_SOURCE}
+##############################################
+# prebuild filecoin-ffi
+COPY extern /go/src/extern
+COPY build /go/src/build
+COPY .git/modules/extern/filecoin-ffi /go/src/.git/modules/extern/filecoin-ffi
+
+RUN make build/.filecoin-install
+##############################################
+COPY . /go/src
+##############################################
+
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+      make debug
+#########################################################################################
+FROM ubuntu:20.04 as runner
+
+RUN apt update && apt install -y \
+      curl \
+      hwloc \
+      jq
+
+## Fix missing lib libhwloc.so.5
+RUN ls -1 /lib/*/libhwloc.so.* | head -n 1 | xargs -n1 -I {} ln -s {} /lib/libhwloc.so.5
+
+WORKDIR /app
+
+#########################################################################################
+FROM runner as boost-main
+
+ARG BUILD_VERSION=0.1
+
+LABEL org.opencontainers.image.version=$BUILD_VERSION \
+      org.opencontainers.image.authors="Boost Dev Team" \
+      name="boost-main" \
+      maintainer="Boost Dev Team" \
+      vendor="Boost Dev Team" \
+      version=$BUILD_VERSION \
+      release=$BUILD_VERSION \
+      summary="This image is used to host the boost-dev storage provider" \
+      description="This image is used to host the boost-dev storage provider"
+
+ENV BOOST_PATH /var/lib/boost
+VOLUME /var/lib/boost
+EXPOSE 8080
+
+COPY --from=builder /go/src/boost /usr/local/bin/
+COPY docker/boost-client/entrypoint.sh /app/
+COPY docker/boost-client/sample/* /app/sample/
+RUN boost -v
```

Comment:
We can probably get rid of this line. This command is not a deamon or long running process. I think container will exit after this. Did we test this Dockerfile?

Reply:
For line 68 - we can't remove this line cause this is the actual boost binary.
For line 69 -  It will exit but I think it's better to have the entrypoint still and just put `boost -v` to confirm that the binary is there and it's on the env path. 

```
#!/usr/bin/env bash
set -e
boost -v
```

IMO, we can remove 70 and 71.

The container will exit but the point is to create the image that can be used to build an containerized environment rather than a running process.

Yes this was tested, here's the output (verification) after running it on my local dockerhub.
<img width="1259" alt="image" src="https://github.com/filecoin-project/boost/assets/4479171/7f986dab-3af8-48cf-a4b7-437daea65ba3">